.github/workflows/dev-setup.yml

```diff
@@ -0,0 +1,90 @@
+name: Dev environment setup
+
+on:
+  push:
+    branches: [master]
+  workflow_dispatch:
+    inputs:
+      repository:
+        description: >
+          [Optional] Repository name with owner. For example, mlflow/mlflow.
+           Defaults to the repository that triggered a workflow.
+        required: false
+        default: ""
+      ref:
+        description: >
+          [Optional] The branch, tag or SHA to checkout. When checking out the repository that
+           triggered a workflow, this defaults to the reference or SHA for that event. Otherwise,
+           uses the default branch.
+        required: false
+        default: ""
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref == 'refs/heads/master' && github.run_number || github.ref }}
+  cancel-in-progress: true
+
+defaults:
+  run:
+    shell: bash --noprofile --norc -exo pipefail {0}
+
+jobs:
```

Comment:
Let's use `strategy.matrix` to DRY the code: https://github.com/mlflow/mlflow/blob/8b0bfc5e9ce46545d452181dc181896043eb8088/.github/workflows/js.yml#L23

Reply:
great idea!